lib/SymbolGraphGen/Edge.cpp

```diff
@@ -36,6 +36,15 @@ const ValueDecl *getForeignProtocolRequirement(const ValueDecl *VD, const Module
     requirements.pop();
   }
 }
+
+const ValueDecl *getProtocolRequirement(const ValueDecl *VD) {
+  auto reqs = VD->getSatisfiedProtocolRequirements();
+
+  if (!reqs.empty())
+    return reqs.front();
```

Comment:
Is it safe to pick the first one? A member can satisfy multiple protocol requirements, e.g.:

```swift
protocol P {
    func foo()
}

protocol Q {
    func foo()
}

struct S: P, Q {
    func foo() {}
}
```

Reply:
At the point where this function is being called, all the protocols that are being implemented are on equal footing: None of them have docs, all of them are local to the module. Should we change the format of `sourceOrigin` to account for multiple protocols?

Reply:
I think this might be something worth considering for the future (if we have a driver for it), but this isn't blocking this PR.